charts/newrelic-infrastructure-beta/templates/_helpers.tpl

```diff
@@ -0,0 +1,182 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "newrelic.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "newrelic.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+
+{{/* Generate mode label */}}
+{{- define "newrelic.mode" }}
+{{- if .Values.privileged -}}
+privileged
+{{- else -}}
+unprivileged
+{{- end }}
+{{- end -}}
+
+{{/* Common labels */}}
+{{- define "newrelic.labels" -}}
+helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+app.kubernetes.io/name: {{ include "newrelic.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+mode: {{ template "newrelic.mode" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+
+{{/* Create the name of the service account to use */}}
+{{- define "newrelic.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "newrelic.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+Return the cluster name
+*/}}
+{{- define "newrelic.cluster" -}}
+{{- if .Values.cluster -}}
+  {{- .Values.cluster -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.cluster -}}
+    {{- .Values.global.cluster -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return local licenseKey if set, global otherwise
+*/}}
+{{- define "newrelic.licenseKey" -}}
+{{- if .Values.licenseKey -}}
+  {{- .Values.licenseKey -}}
+{{- else if .Values.global -}}
+  {{- if .Values.global.licenseKey -}}
+    {{- .Values.global.licenseKey -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the name of the secret holding the License Key
+*/}}
+{{- define "newrelic.licenseCustomSecretName" -}}
+{{- if .Values.customSecretName -}}
+  {{- .Values.customSecretName -}}
+{{- else if and .Values.global -}}
```

Comment:
Maybe here there is no possibility of doing the above proposed change. But why have you used a `and` and then put another `if` in other line?

```suggestion
{{- else if (and .Values.global .Values.global.customSecretName) -}}
```